dlc/src/util.rs

```diff
@@ -199,10 +200,14 @@ pub fn sign_multi_sig_input<C: Signing>(
 }
 
 /// Transforms a redeem script for a p2sh-p2w* output to a script signature.
-pub(crate) fn redeem_script_to_script_sig(redeem: &Script) -> Script {
+pub(crate) fn redeem_script_to_script_sig(redeem: &Script) -> ScriptBuf {
     match redeem.len() {
-        0 => Script::new(),
-        _ => Builder::new().push_slice(redeem.as_bytes()).into_script(),
+        0 => ScriptBuf::new(),
+        _ => {
+            let mut bytes = PushBytesBuf::new();
+            bytes.extend_from_slice(redeem.as_bytes()).unwrap();
+            ScriptBuf::new_witness_program(&WitnessProgram::new(WitnessVersion::V1, bytes).unwrap())
```

Comment:
witness v1 is taproot, this doesn't seem correct

Reply:
Thanks fixed (was copy pasted from my taproot branch...)